pkg/api-server/resource_endpoints.go

```diff
@@ -147,6 +148,22 @@ func (r *resourceEndpoints) updateResource(ctx context.Context, res model.Resour
 	}
 }
 
+func (r *resourceEndpoints) addCreateOrUpdateEndpointReadOnly(ws *restful.WebService, pathPrefix string) {
+	ws.Route(ws.PUT(pathPrefix+"/{name}").To(r.createOrUpdateResourceReadOnly).
+		Doc("Not allowed in read-only mode.").
+		Returns(http.StatusMethodNotAllowed, "Not allowed in read-only mode.", restful.ServiceError{}))
+}
+
+func (r *resourceEndpoints) createOrUpdateResourceReadOnly(request *restful.Request, response *restful.Response) {
+	err := response.WriteErrorString(http.StatusMethodNotAllowed,
+		"On Kubernetes you cannot change the state of Kuma resources with 'kumactl apply'."+
+			" As a best practice, you should be using 'kubectl apply' instead."+
+			" You can still use 'kumactl' to make read-only operations. On Universal this limitation does not apply.\n")
```

Comment:
Please update the error message to this slightly different version that takes into account that the state change could have been sent via `kumactl` or by using the HTTP API directly:

> on Kubernetes you cannot change the state of Kuma resources with "kumactl apply" or via the HTTP API and as a best practice you should always be using "kubectl apply" instead; you can still use kumactl or the HTTP API to make read-only operations. On Universal this limitation does not apply.

Reply:
Done, thanks!